src/Lib.P2P.Tests/Transports/TcpTest.cs

```diff
@@ -0,0 +1,232 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Lib.P2P.Transports;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using MultiFormats;
+
+namespace Lib.P2P.Tests.Transports
+{
+    [TestClass]
+    public class TcpTest
+    {
+        [TestMethod]
+        public void Connect_Unknown_Port()
+        {
+            var tcp = new Tcp();
+            ExceptionAssert.Throws<SocketException>(() =>
+            {
+                var _ = tcp.ConnectAsync("/ip4/127.0.0.1/tcp/32700").Result;
+            });
+        }
+
+        [TestMethod]
+        public void Connect_Missing_TCP_Port()
+        {
+            var tcp = new Tcp();
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = tcp.ConnectAsync("/ip4/127.0.0.1/udp/32700").Result;
+            });
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = tcp.ConnectAsync("/ip4/127.0.0.1").Result;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)